### DISCUSSION TITLE
FAST checkpoints access

### DISCUSSION THREAD

**Discussion by josorio1** [mindee/doctr#1541] [General]
Hi! I see here ([https://github.com/mindee/doctr/blob/0d849152a852cb55b3fb0cc0e5e602600349d97d/doctr/models/detection/fast/pytorch.py](url)) that FAST checkpoints URLs are already available, but when I try to download the weights the request is denied / doesn't work.

I am not sure if this is a bug or if the weights are actually not available yet, so I just wanted to ask this.

Thanks in advance

**Comment by felixdittrich92** [comment]
CC @odulcy-mindee i have had the same today :)

**Reply by odulcy-mindee** [reply]
@josorio1 Thanks for notifying us, this is not expected. We're on it

**Reply by odulcy-mindee** [reply]
@josorio1 It's now fixed :ok_hand:

**Reply by felixdittrich92** [reply]
Thanks 😅

**Reply by josorio1** [reply]
Wow that was FAST. Thank you very much guys, I love your work!

**Reply by felixdittrich92** [reply]
@josorio1 If you test the fast models atm we would be happy about a feedback :)
I have heard already that the fast models works a bit better on straight pages compared to db_resnet50 and much better on rotated samples with `assume_straight_pages=False`

**Reply by josorio1** [reply]
@felixdittrich92 so far your pretained weights seem to be working well, although I can't tell yet if they are worse or better than db_resnet50. However, I have also tried to use the weights provided in the original repository of FAST ([https://github.com/czczup/FAST)]) and I'm not getting any detections. The code itself seems to be working properly, so I am a bit puzzled. Do you know if this is normal?

**Reply by felixdittrich92** [reply]
Yeah our implementation differs a bit (different layer names / backbone is wrapped in a sequential).
It is possible but you would need to modify the original state_dict to match with our implementation :)
